keygen/src/keygen.rs

```diff
@@ -73,6 +73,66 @@ fn output_keypair(
     Ok(())
 }
 
+fn grind_validator_starts_with(v: String) -> Result<(), String> {
+    if v.matches(":").count() != 1 || (v.starts_with(":") || v.ends_with(":")) {
+        return Err(String::from("Expected : between PREFFIX and COUNT"));
+    }
+    let args: Vec<&str> = v.split(':').collect();
+    let s = bs58::decode(&args[0]).into_vec()
+    .map(|_| ())
+    .map_err(|err| format!("{}: {:?}", args[0], err));
+    if s.is_err() {
```

Comment:
Instead of checking explicitly for `if s.is_err() { ...`, try this:

```
bs58::decode(&args[0]).into_vec().map_err(|err| format!("{}: {:?}", args[0], err))?;
```

Reply:
updated